server/game/cards/07-GR/Technorachnophobia.js

```diff
@@ -0,0 +1,34 @@
+const Card = require('../../Card.js');
+
+class Technorachnophobia extends Card {
+    // Play: Discard the top 10 cards of your deck. If you discard 5
+    // or more cards of the same house this way, steal 2.
+    setupCardAbilities(ability) {
+        this.play({
+            gameAction: ability.actions.discard((context) => ({
+                target: context.player.deck.slice(0, Math.min(context.player.deck.length, 10))
+            })),
+            then: {
+                condition: (context) =>
+                    context.preThenEvents &&
+                    context.preThenEvents
+                        .flatMap((event) => event.card.getHouses())
+                        .filter((h, i, a) => a.indexOf(h) === i) // get the list of unique houses
```

Comment:
I wonder if we can do it in less lines:

context.preThenEvents
  .flatMap((event) => event.card.getHouses())
  .reduce((obj, str) => (obj[str] = (obj[str] || 0) + 1, obj[str] >= 5) ? true : obj, {}) === true

ChatGPT (TM)

Reply:
The code didn't work but I tweaked it a little to make it work. I'm not sure it's more readable this way but I'm fine either way.  Please take another look.